### PR TITLE
Upgrading GATK formula and bumping version to 4.1.3.0

### PR DESCRIPTION
**Pull request by lbergelson**
I've been maintaining my own version of a GATK formula for internal use here: https://github.com/broadinstitute/hombrew-dsp

I figured it might make sense to merge them since I think my formula has a few additional features that are nice.

* Updating the GATK formula to match broadinstitute/hombrew-dsp
* This includes some additional features:
  * Additional helper scripts are installed
  * Adds a stronger test case that actually executes a very simple tool
  * Adds a caveats section to explain that python / R dependencies for some tools are not installed as part of this.

* Update version from 4.1.2.0 -> 4.1.3.0

- [x] Have you followed the [guidelines for contributing](https://github.com/brewsci/homebrew-bio/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/brewsci/homebrew-bio/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source FORMULA`, where `FORMULA` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict FORMULA` (after doing `brew install FORMULA`)?

-----

### CODE REVIEW

File: Formula/gatk.rb [hunk 1]
```diff
@@ -1,22 +1,44 @@
...
+    See the GATK readme for detailed installation instructions.
+       https://github.com/broadinstitute/gatk
+
+    The recommended way of running the tools with complex python dependencies is to use the pre-built docker images instead of attempting to install them locally.
+    Gatk dockers are available on docker hub:
+       https://hub.docker.com/r/broadinstitute/gatk/tags/
+  EOS
+  end
   test do
-    assert_match "Usage", shell_output("#{bin}/gatk --help 2>&1")
```
Comment (tseemann) [inline Formula/gatk.rb]: Why did you remove this test line?

File: Formula/gatk.rb [hunk 2]
```diff
@@ -1,22 +1,44 @@
...
-  sha256 "ffc5f9b3d4b35772ee5dac3060b59dc657f30e830745160671d84d732c30dc65"
-  head "https://github.com/broadinstitute/gatk.git"
+  homepage "https://software.broadinstitute.org/gatk"
+  url "https://github.com/broadinstitute/gatk/releases/download/4.1.3.0/gatk-4.1.3.0.zip"
+  sha256 "56fd4f03b15a8a01eaa4629f62e3ab15e4d4b957c787efd2d5629b2658c3df0a"
 
   bottle :unneeded
 
-  depends_on :java
+  depends_on :java => "1.8"
```
Comment (tseemann) [inline Formula/gatk.rb]: Does linuxbrew honour the Java version @sjackman ?

File: Formula/gatk.rb (hunk 1 above)
Comment (lbergelson) [reply Formula/gatk.rb]: Oh, you're right.  I copied in my test, but better to keep your test as well.

Comment (lbergelson) [reply Formula/gatk.rb]: Oh, you're right.  I copied in my test, but better to keep your test as well.  I'll add that back.

**Comment by lbergelson** [comment]
The tests are failing but I don't understand why

```
==> brew test brewsci/bio/gatk --verbose
mv: rename brew-test-bot.xml to brew-test-bot-macos.xml: No such file or directory
Exited with code 1
```

When I run `brew test Formula/gatk.rb` it works locally.

**Comment by tseemann** [comment]
@lbergelson i am getting that error too, i think it is affecting all builds. i am hoping @sjackman will figure it out now he is back from camping :-P

**Comment by tseemann** [comment]
@lbergelson can you rebase and re-push?

```
git rebase master
git commit --amend gatk.rb
git push -f
```

**Comment by lbergelson** [comment]
@tseeman Rebased

**Comment by tseemann** [comment]
Thank you @lbergelson - more tests always welcomed!

**Comment by lbergelson** [comment]
@tseemann Thank you!
